test/unit/foreman/renderer/scope/macros/base_test.rb

```diff
@@ -59,6 +59,34 @@ class BaseMacrosTest < ActiveSupport::TestCase
     assert_equal '', @scope.pxe_kernel_options
   end
 
+  describe '#host_kernel_release' do
+    test 'should return kernel release' do
+      host = FactoryBot.create(:host)
+      fact = FactoryBot.create(:fact_name, name: 'kernelrelease')
+      FactoryBot.create(:fact_value, fact_name: fact, host: host, :value => '1.2.3')
+      assert_equal '1.2.3', @scope.host_kernel_release(host)
+    end
+
+    test 'should return nil if no kernel release fact is available' do
+      host = FactoryBot.create(:host)
+      assert_nil @scope.host_kernel_release(host)
+    end
+
+    test 'should return kernel release and based on backup facts even if there are multiple other facts' do
+      host = FactoryBot.create(:host)
+      fact1 = FactoryBot.create(:fact_name, name: 'ansible_kernel', :type => 'FactName::Ansible')
```

Comment:
Perhaps a more descriptive name would be good (`ansible_kernel_fact` perhaps?)

Reply:
updated fact names, hope it's ready now